main.cpp

```diff
@@ -18,13 +18,18 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+    Rewriter& refRewriter;
+
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : refRewriter(rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult& Result) override {
+        auto cStyleCastExpr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+        if (cStyleCastExpr) {
+            refRewriter.ReplaceText(cStyleCastExpr->getLParenLoc(), 1, "static_cast<");
+            refRewriter.ReplaceText(cStyleCastExpr->getRParenLoc(), 1, ">(");

```

Comment:
Please describe how to change the program code to get the following result:
```cpp
int main() {
    float fooooo;
    int i = static_cast<int>          (fooooo)      ;
    return 0;
}
```
 for the origin `test.cpp`:
```cpp
int main() {

    float fooooo;

    int i = (int)          fooooo      ;

    return 0;

}
```

Reply:
Since the test code has changed and spaces have been added there, it is worth reviewing the function of inserting an opening bracket. Remove the opening bracket from refRewriter.ReplaceText(cStyleCastExpr->getRParenLoc(), 1, ">(");
Now we need to retrieve the cast subexpression as it was written in the source code, this can be obtained thanks to cStyleCastExpr->getSubExprAsWritten(), then from the resulting Expr go to its first token and insert the opening bracket
rewriter.InsertText(cStyleCastExpr->getSubExprAsWritten()->getBeginLoc(), "(");